autocorpus/autocorpus.py

```diff
@@ -265,7 +323,20 @@ def __merge_table_data(self):
                             }
                         )
 
-    def process_files(self):
+    def __process_supplementary_file(self, file: Path):
+        match file.suffix:
+            case ".html" | ".htm":
+                self.__process_html_article(file)
+            case ".xml":
+                pass
+            case ".pdf":
+                is_extracted = self.__extract_pdf_content(file)
+                if not is_extracted:
+                    logger.info(f"Unable to extract text/tables from {file.name}")
```

Comment:
Not needed if you follow my suggestion above:

```suggestion
                self.__extract_pdf_content(file)
```

Reply:
The data models aren't necessarily a guarantee that content will be extracted, the file provided could simply not have content they're capable of extracting or are corrupt, but that shouldn't really cause the whole execution to stop (null/None reference exception?)

Reply:
Refactoring now as it currently doesn't handle these scenarios well, I have removed the bool check & log from the match case, its already logging inside the extract_pdf_content method